Examples/ObjectiveC/ManageOfflineRegionsExample.m

```diff
@@ -3,12 +3,281 @@
+    }
+
+    // Reload the table to update the progress percentage for each offline pack.
+    [self.tableView reloadData];
+
+}
+
+#pragma mark - UITableView configuration
+
+// Create the table view which will display the downloaded regions.
+- (NSInteger)tableView:(nonnull UITableView *)tableView
+ numberOfRowsInSection:(NSInteger)section {
+    if (MGLOfflineStorage.sharedOfflineStorage.packs.count != 0) {
+        return MGLOfflineStorage.sharedOfflineStorage.packs.count;
+    } else {
+        return 0;
+    }
+
+}
+
+- (UIView *)tableView:(UITableView *)tableView
+viewForHeaderInSection:(NSInteger)section {
+    UILabel * label = [[UILabel alloc] init];
+    label.backgroundColor = UIColor.systemBlueColor;
+    label.textColor = UIColor.whiteColor;
+    label.font = [UIFont preferredFontForTextStyle: UIFontTextStyleHeadline];
+    label.textAlignment = NSTextAlignmentCenter;
+
+
+    if (MGLOfflineStorage.sharedOfflineStorage.packs != nil) {
+        label.text = @"Offline maps";
+    } else {
+        label.text = @"No offline maps";
+    }
+
+    return label;
+}
+
+- (CGFloat)tableView:(UITableView *)tableView
+heightForHeaderInSection:(NSInteger)section {
+    return 50.0;
+}
+
+- (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView
+                 cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath {
+
+    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cell" forIndexPath:indexPath];
+
+    [cell initWithStyle:UITableViewCellStyleSubtitle reuseIdentifier:@"cell"];
+    
+    if (MGLOfflineStorage.sharedOfflineStorage.packs.count != 0 )  {
+        MGLOfflinePack * pack = MGLOfflineStorage.sharedOfflineStorage.packs[indexPath.row];
+        long number = indexPath.row + 1;
+        long countOfBytesCompleted = MGLOfflineStorage.sharedOfflineStorage.packs[indexPath.row].progress.countOfBytesCompleted;
+        NSByteCountFormatter *formattedCountOfBytedCompleted = [NSByteCountFormatter stringFromByteCount:countOfBytesCompleted countStyle:NSByteCountFormatterCountStyleMemory];
+        float countOfResourcesCompleted = pack.progress.countOfResourcesCompleted;
+        float countOfResourcesExpected = pack.progress.countOfResourcesExpected;
+        
+        float percentCompleted = ((countOfResourcesCompleted)/(countOfResourcesExpected) * 100);
+
+        cell.textLabel.text = [NSString stringWithFormat:@"Region %ld: size: %@", number, formattedCountOfBytedCompleted];
+
+        cell.detailTextLabel.text = [NSString stringWithFormat:@"Percent completion: %.1f%%", percentCompleted];
+
+    }
+
+    return cell;
+}
+
+-(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
+    MGLTilePyramidOfflineRegion *selectedRegion = MGLOfflineStorage.sharedOfflineStorage.packs[indexPath.row].region;
+
+    if ([selectedRegion isKindOfClass:[MGLTilePyramidOfflineRegion class]]) {
+        [_mapView setVisibleCoordinateBounds:selectedRegion.bounds animated:YES];
```

Comment:
Please use the `self.mapView` style of setting all properties instead of `_mapView`, which is the syntax we use across all ObjC examples. (You can read more about [here](https://stackoverflow.com/questions/14112715/self-variablename-vs-variablename-vs-sysnthesize-variablename) and [here](https://teamtreehouse.com/community/objc-underscore-self-notation-difference)).

This should also allow you to remove all your `@synthesize` calls.